src/i18n.ts

```diff
@@ -0,0 +1,23 @@
+function loadResourcesFromFile() {
+  const lang = window.i18next.language;
+  import(`../translations/${lang}.json`)
+    .then((json) => {
+      const data = json ?? {};
+      window.i18next.addResourceBundle(lang, '@openmrs/openmrs-form-engine-lib', data);
+    })
+    .catch((err) => console.error(err));
+}
+
+function setupFormEngineLibI18n() {
+  loadResourcesFromFile();
+  window.i18next.on('languageChanged', loadResourcesFromFile);
+
+  const languageChangeObserver = new MutationObserver(loadResourcesFromFile);
+
+  languageChangeObserver.observe(document.documentElement, {
+    attributeFilter: ['lang'],
+    attributes: true,
+  });
+}
+
+setupFormEngineLibI18n();
```

Comment:
Instead of having this implicitly loaded, this should be explicitly called as part of the form engine initialization.